src/openms/source/FORMAT/MzTab.cpp

```diff
@@ -1634,769 +1677,858 @@ namespace OpenMS
+    return mztab;
+  }
 
-      const vector<PeptideIdentification>& pep_ids = f.getPeptideIdentifications();
-      for (PeptideIdentification const & pep_id : pep_ids)
-      {
-        for (PeptideHit const & hit : pep_id.getHits())
-        {
-          vector<String> ph_keys;
-          hit.getKeys(ph_keys);
-          for (String& s : ph_keys)
-          {
-            if (s.has(' '))
-            {
-              s.substitute(' ', '_');
-            }
-          }
-          peptide_hit_user_value_keys.insert(ph_keys.begin(), ph_keys.end());
-        }
-      }
-    }
+  MzTabPeptideSectionRow MzTab::peptideSectionRowFromFeature_(
+    const Feature& f, 
+    const set<String>& feature_user_value_keys,
+    const set<String>& peptide_hit_user_value_keys,
+    const vector<String>& fixed_mods)
+  {
+    MzTabPeptideSectionRow row;
+    row.mass_to_charge = MzTabDouble(f.getMZ());
+    MzTabDoubleList rt_list;
+    vector<MzTabDouble> rts;
+    rts.emplace_back(MzTabDouble(f.getRT()));
+    rt_list.set(rts);
+    row.retention_time = rt_list;
 
-    MzTabPeptideSectionRows rows;
-    for (Size i = 0; i < feature_map.size(); ++i)
+    // set rt window if a bounding box has been set
+    vector<MzTabDouble> window;
+    if (f.getConvexHull().getBoundingBox() != DBoundingBox<2>())
     {
-      MzTabPeptideSectionRow row;
-      const Feature& f = feature_map[i];
-      row.mass_to_charge = MzTabDouble(f.getMZ());
-      MzTabDoubleList rt_list;
-      vector<MzTabDouble> rts;
-      rts.emplace_back(MzTabDouble(f.getRT()));
-      rt_list.set(rts);
-      row.retention_time = rt_list;
-
-      // set rt window if a bounding box has been set
-      vector<MzTabDouble> window;
-      if (f.getConvexHull().getBoundingBox() != DBoundingBox<2>())
-      {
-        window.emplace_back(MzTabDouble(f.getConvexHull().getBoundingBox().minX()));
-        window.emplace_back(MzTabDouble(f.getConvexHull().getBoundingBox().maxX()));
-      }
-
-      MzTabDoubleList rt_window;
-      rt_window.set(window);
-      row.retention_time_window = rt_window;
-      row.charge = MzTabInteger(f.getCharge());
-      row.peptide_abundance_stdev_study_variable[1];
-      row.peptide_abundance_std_error_study_variable[1];
-      row.peptide_abundance_study_variable[1] = MzTabDouble(f.getIntensity());
-      row.best_search_engine_score[1] = MzTabDouble();
-      row.search_engine_score_ms_run[1][1] = MzTabDouble();
-
-      // create opt_ column for peptide sequence containing modification
-      MzTabOptionalColumnEntry opt_global_modified_sequence;
-      opt_global_modified_sequence.first = String("opt_global_modified_sequence");
-      row.opt_.push_back(opt_global_modified_sequence);
-
-      // create and fill opt_ columns for feature (peptide) user values
-      addMetaInfoToOptionalColumns(feature_user_value_keys, row.opt_, String("global"), f);
+      window.emplace_back(MzTabDouble(f.getConvexHull().getBoundingBox().minX()));
```

Comment:
do you need MzTabDouble when emplacing?

Reply:
I think so because the constructor is explicit: explicit MzTabDouble(const double v);